mood-analysis.rb

```diff
@@ -28,5 +28,65 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+# puts analyze_mood(text[0])
+# puts analyze_mood(text[1])
+
+def string_analysis(array)
+  array.each do |day|
+    p array[0][0..4] + analyze_mood(day)
+  end
+end
+
+# string_analysis(text)
+# 3. Write a method called `happy_days` to determine how many logged entries it takes until there have been three :-) happy days.
+def happy_days(words)
+  count =[]
+  days = 0
+  words.each do |day|
+    moods = analyze_mood(day)[-3..-1]
+    count << moods.include?(":-)")
+      days += 1
+    if count.length == 3
+      puts "It takes #{days} entries for 3 happy days to occur"
+      break
+    end
+  end
+  puts "sorry, to many moods!"
+end
+
+# happy_days(text)
+
+# 4. Write a method called `overall_mood` to determine the most common mood across all logged entries.
+
+def overall_mood(words)
+  happy = []
+  sad = []
+  neutral = []
+
+  words.each do |day|
+    moods = analyze_mood(day)[-3..-1]
+    if moods.include?(":-)")
+      happy << moods
+    elsif moods.include?(":-(")
+      sad << moods
+    else
+      neutral << moods
+    end
+  end
+
+    p happy
+    p sad
+    p neutral
+
+      if happy.length > sad.length && neutral.length
+        puts "The most common mood is :-)"
+          return happy
+      elsif sad.length > happy.length && neutral.length
+        puts "The most common mood is :-("
+          return sad
+      else
+        puts "The most common mood is :-|"
+          return neutral
+      end
+end
```

Comment:
Since you are comparing by length of the arrays, again here, I think it makes more sense to just have straight counters instead of pushing each day into and array and then finding the length on those arrays. This way works, so that's good, but it's not as expandable or efficient.

For comparison, here's my overall_mood method:
```
def overall_mood(entries)
  return nil if entries.length == 0
  emoticons = Hash.new(0)
  entries.each do |entry|
    emoticon = analyze_mood(entry)
    emoticons[emoticon] += 1
  end
  return emoticons.max_by{|k,v| v}[0]
end
```